roles/openshift_facts/library/openshift_facts.py

```diff
@@ -394,6 +394,31 @@ def set_master_selectors(facts):
             if 'registry_selector' not in facts['master']:
                 facts['master']['registry_selector'] = selector
     return facts
+    
+def set_global_proxy(facts):
+    """ Set global proxy configuration options
+        Args:
+            facts (dict): existing facts
+        Returns:
+            facts(dict): Ammended facts with those not present
+
+    """
+    if 'common' in facts:
+        if 'http_proxy' not in facts['common']:
+            facts['common']['http_proxy'] = ''
+        if 'https_proxy' not in facts['common']:
+            facts['common']['https_proxy'] = ''
+        if 'noproxy' not in facts['common']:
+            facts['common']['noproxy'] = ''
+    
+    if 'master' in facts:
+        if 'builddefaults_http_proxy' not in facts['master']:
+            facts['master']['builddefaults_http_proxy'] = facts['common']['http_proxy']
+        if 'builddefaults_https_proxy' not in facts['master']:
+            facts['master']['builddefaults_https_proxy'] = facts['common']['https_proxy']
+        if 'builddefaults_noproxy' not in facts['master']:
+            facts['master']['builddefaults_noproxy'] = facts['common']['noproxy']
+    return facts
```

Comment:
Since these do not require any computation, we should be adding these to the default facts in get_defaults() instead.